speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionMakerTests.cpp

```diff
@@ -55,3 +60,91 @@ TEST_CASE(DecisionMakerTests, ChangeSettingsTestNoIntervention, ChangeSettingsTe
 TEST_CASE(DecisionMakerTests, ChangeSettingsTestAlwaysIntervene, ChangeSettingsTest, (INTERVENTION_TYPE_COMPLETE_TAKEOVER));
 TEST_CASE(DecisionMakerTests, ChangeSettingsTestIndication, ChangeSettingsTest, (INTERVENTION_TYPE_ONLY_SIGNALS));
 TEST_CASE(DecisionMakerTests, ChangeSettingsTestPerformWhenNeeded, ChangeSettingsTest, (INTERVENTION_TYPE_SHARED_CONTROL));
+
+/// @brief				 Tests if the decision maker can be initialized
+/// @param  p_decisionMaker the decision maker that will be initialized
+void InitializeTest(TDecisionMaker& p_decisionMaker)
+{
+    tCarElt car;  // need data
+    tSituation situation;
+    situation.deltaTime = 108;
+    car.pub.speed = 144;
+    tTrack track;
+    track.filename = "trackfile";
+    track.name = "track_1";
+    track.version = 0;
+
+    p_decisionMaker.Config.SetUserId("1");
+
+    std::string findfilepath = "speed-dreams\\" ROOT_FOLDER "\\data\\blackbox";
+    std::cout << SD_DATADIR_SRC << std::endl;
+    chdir(SD_DATADIR_SRC);
+    ASSERT_TRUE(FindFileDirectory(findfilepath, "blackbox.exe"));
+    std::string bbPath = findfilepath.append("\\blackbox.exe");
+    p_decisionMaker.Initialize(&car, &situation, &track, bbPath, true);
```

Comment:
Does the same, but uses less memory
```suggestion
    findfilepath.append("\\blackbox.exe");
    p_decisionMaker.Initialize(&car, &situation, &track, findfilepath, true);
```